### ISSUE TITLE
neat read-simulator takes very long

### ISSUE THREAD

**Issue by Npaffen** [ncsa/NEAT#80]
**Describe the bug**
When I try to simulate reads with vcf-input the process seems to take forever. I did the following :
`
 neat read-simulator -c neat_config.yaml -o neat/`

```
2023-07-13 17:00:37,523:INFO:neat.read_simulator.runner:Using configuration file neat_config.yaml
2023-07-13 17:00:37,523:INFO:neat.read_simulator.runner:Saving output files to data
2023-07-13 17:00:37,524:INFO:neat.read_simulator.utils.options:Run Configuration...
2023-07-13 17:00:37,524:INFO:neat.read_simulator.utils.options:Input fasta: chr22.fa
2023-07-13 17:00:37,524:INFO:neat.read_simulator.utils.options:Producing the following files:
  - data/neat_r1.fastq.gz
- data/neat_r2.fastq.gz
- data/neat_golden.bam

2023-07-13 17:00:37,524:INFO:neat.read_simulator.utils.options:Single threading - 1 thread.
2023-07-13 17:00:37,524:INFO:neat.read_simulator.utils.options:Running in paired-ended mode.
2023-07-13 17:00:37,524:INFO:neat.read_simulator.utils.options:Generating fragment model based on mean=300.0, st dev=30.0
2023-07-13 17:00:37,524:INFO:neat.read_simulator.utils.options:Using a read length of 126
2023-07-13 17:00:37,524:INFO:neat.read_simulator.utils.options:Average coverage: 1
2023-07-13 17:00:37,524:INFO:neat.read_simulator.utils.options:Using default error model.
2023-07-13 17:00:37,524:INFO:neat.read_simulator.utils.options:Ploidy value: 2
2023-07-13 17:00:37,524:INFO:neat.read_simulator.utils.options:RNG seed value for run: 5404759810307010
2023-07-13 17:00:37,524:INFO:neat.read_simulator.runner:Reading Models...
2023-07-13 17:00:37,525:INFO:neat.read_simulator.runner:Reading chr22.fa.
2023-07-13 17:00:40,037:INFO:neat.read_simulator.runner:Beginning simulation.
2023-07-13 17:00:40,584:INFO:neat.read_simulator.runner:Generating variants for chr22
2023-07-13 17:03:39,927:INFO:neat.read_simulator.utils.generate_variants:Finished generating random mutations in 2.98 minutes
2023-07-13 17:03:39,927:INFO:neat.read_simulator.utils.generate_variants:Added 51203 mutations to chr22
2023-07-13 17:03:39,927:INFO:neat.read_simulator.runner:Outputting temp vcf for chr22 for later use
2023-07-13 17:03:40,358:INFO:neat.read_simulator.utils.local_file_writer:Finished outputting temp vcf/fasta
2023-07-13 17:03:40,361:INFO:neat.read_simulator.utils.generate_reads:Sampling reads..

```

All the files to replicate the issue can be downloaded [here](https://we.tl/t-oyBw2GiCFZ)

**Comment by joshfactorial** [comment]
All right, I will check into this issue.

**Comment by a00101** [comment]
I got same problem.
Infinite time consuming,, neat.read_simulator.utils.generate_reads:Sampling reads...

**Comment by joshfactorial** [comment]
It'll be slow for large files. A stopgap solution is to break files up and run jobs concurrently.

**Comment by joshfactorial** [comment]
Apologies, everybody. For now, NEAT is simply slow. We are working on some optimizations for future releases. Try breaking large files or large runs (high coverage, high mutation rates, high ploidy) into several runs and run them in parallel then combine the results. We will work on our end to implement multithreading or some similar solution. Thank you for your patience.

**Comment by abbyevewilliams** [comment]
Just a thought on splitting the genome into multiple files as suggested above. I'm working with a very large genome and have split it into 30 chunks. Because NEAT now needs a config file, in theory this means I need a separate config file for each run. For this reason it would be useful to be able to specify options in the command line as well as in the config file - a separate issue, but I thought worth mentioning in this context.

**Comment by joshfactorial** [comment]
True. We can write some utilities to break up genomes and reassemble, similar to what was available in NEAT 2.1, as a stopgap. It could output matching yml files/streams. If you prefer to use CLI options to split things up, NEAT 3.4 uses CLI and matches functionality closest to the original (which is also still available as Neat 2.1). I can start a separate issue for this.
